python-fire: build error caused by unified diff test failure

This might just be darwin; haven't tried on Linux.

    $ nix-shell -p python37Packages.fire
    these derivations will be built:
      /nix/store/dpcwgrlydrrlbdrnf0mi9hmr0kjvia89-python3.7-fire-0.1.3.drv
    building '/nix/store/dpcwgrlydrrlbdrnf0mi9hmr0kjvia89-python3.7-fire-0.1.3.drv'...
    unpacking sources
    unpacking source archive /nix/store/20rzw7yi3yvjb7w2ixzj6aqa5fim6p1q-source
    source root is source
    setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/setup.py
    patching sources
    configuring
    building
    running bdist_wheel
    The [wheel] section is deprecated. Use [bdist_wheel] instead.
    running build
    running build_py
    creating build
    creating build/lib
    creating build/lib/fire
    copying fire/fire_import_test.py -> build/lib/fire
    copying fire/interact_test.py -> build/lib/fire
    copying fire/trace.py -> build/lib/fire
    copying fire/test_components.py -> build/lib/fire
    copying fire/decorators_test.py -> build/lib/fire
    copying fire/test_components_test.py -> build/lib/fire
    copying fire/helputils_test.py -> build/lib/fire
    copying fire/trace_test.py -> build/lib/fire
    copying fire/completion.py -> build/lib/fire
    copying fire/testutils.py -> build/lib/fire
    copying fire/inspectutils.py -> build/lib/fire
    copying fire/fire_test.py -> build/lib/fire
    copying fire/__init__.py -> build/lib/fire
    copying fire/completion_test.py -> build/lib/fire
    copying fire/core.py -> build/lib/fire
    copying fire/parser_test.py -> build/lib/fire
    copying fire/core_test.py -> build/lib/fire
    copying fire/parser.py -> build/lib/fire
    copying fire/parser_fuzz_test.py -> build/lib/fire
    copying fire/test_components_py3.py -> build/lib/fire
    copying fire/testutils_test.py -> build/lib/fire
    copying fire/interact.py -> build/lib/fire
    copying fire/inspectutils_test.py -> build/lib/fire
    copying fire/helputils.py -> build/lib/fire
    copying fire/decorators.py -> build/lib/fire
    installing to build/bdist.macosx-10.6-x86_64/wheel
    running install
    running install_lib
    creating build/bdist.macosx-10.6-x86_64
    creating build/bdist.macosx-10.6-x86_64/wheel
    creating build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/fire_import_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/interact_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/trace.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/test_components.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/decorators_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/test_components_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/helputils_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/trace_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/completion.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/testutils.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/inspectutils.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/fire_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/__init__.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/completion_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/core.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/parser_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/core_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/parser.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/parser_fuzz_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/test_components_py3.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/testutils_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/interact.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/inspectutils_test.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/helputils.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    copying build/lib/fire/decorators.py -> build/bdist.macosx-10.6-x86_64/wheel/fire
    running install_egg_info
    running egg_info
    creating fire.egg-info
    writing fire.egg-info/PKG-INFO
    writing dependency_links to fire.egg-info/dependency_links.txt
    writing requirements to fire.egg-info/requires.txt
    writing top-level names to fire.egg-info/top_level.txt
    writing manifest file 'fire.egg-info/SOURCES.txt'
    reading manifest file 'fire.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'fire.egg-info/SOURCES.txt'
    Copying fire.egg-info to build/bdist.macosx-10.6-x86_64/wheel/fire-0.1.3-py3.7.egg-info
    running install_scripts
    creating build/bdist.macosx-10.6-x86_64/wheel/fire-0.1.3.dist-info/WHEEL
    creating 'dist/fire-0.1.3-py2.py3-none-any.whl' and adding 'build/bdist.macosx-10.6-x86_64/wheel' to it
    adding 'fire/__init__.py'
    adding 'fire/completion.py'
    adding 'fire/completion_test.py'
    adding 'fire/core.py'
    adding 'fire/core_test.py'
    adding 'fire/decorators.py'
    adding 'fire/decorators_test.py'
    adding 'fire/fire_import_test.py'
    adding 'fire/fire_test.py'
    adding 'fire/helputils.py'
    adding 'fire/helputils_test.py'
    adding 'fire/inspectutils.py'
    adding 'fire/inspectutils_test.py'
    adding 'fire/interact.py'
    adding 'fire/interact_test.py'
    adding 'fire/parser.py'
    adding 'fire/parser_fuzz_test.py'
    adding 'fire/parser_test.py'
    adding 'fire/test_components.py'
    adding 'fire/test_components_py3.py'
    adding 'fire/test_components_test.py'
    adding 'fire/testutils.py'
    adding 'fire/testutils_test.py'
    adding 'fire/trace.py'
    adding 'fire/trace_test.py'
    adding 'fire-0.1.3.dist-info/LICENSE'
    adding 'fire-0.1.3.dist-info/METADATA'
    adding 'fire-0.1.3.dist-info/WHEEL'
    adding 'fire-0.1.3.dist-info/top_level.txt'
    adding 'fire-0.1.3.dist-info/RECORD'
    removing build/bdist.macosx-10.6-x86_64/wheel
    installing
    /private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source/dist /private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source
    Processing ./fire-0.1.3-py2.py3-none-any.whl
    Requirement already satisfied: six in /nix/store/qayyh24znx8kixy7wqz6i5416nyljzxk-python3.7-six-1.12.0/lib/python3.7/site-packages (from fire==0.1.3) (1.12.0)
    Installing collected packages: fire
    Successfully installed fire-0.1.3
    /private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source
    post-installation fixup
    strip is /nix/store/3ibh116gnggzgsvrq8vvalmqyipk8cy9-cctools-binutils-darwin/bin/strip
    stripping (with command strip and flags -S) in /nix/store/b96a60g0apqx015sscpv8jywnjd8svcq-python3.7-fire-0.1.3/lib 
    patching script interpreter paths in /nix/store/b96a60g0apqx015sscpv8jywnjd8svcq-python3.7-fire-0.1.3
    running install tests
    ============================= test session starts ==============================
    platform darwin -- Python 3.7.2, pytest-4.2.1, py-1.7.0, pluggy-0.8.1
    hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source/.hypothesis/examples')
    rootdir: /private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source, inifile: setup.cfg
    plugins: hypothesis-4.7.3
    collected 160 items

    examples/cipher/cipher_test.py .                                         [  0%]
    examples/diff/diff_test.py ......                                        [  4%]
    examples/widget/collector_test.py ...                                    [  6%]
    examples/widget/widget_test.py ..                                        [  7%]
    fire/completion_test.py ..............                                   [ 16%]
    fire/core_test.py .............                                          [ 24%]
    fire/decorators_test.py ...........                                      [ 31%]
    fire/fire_import_test.py ...                                             [ 33%]
    fire/fire_test.py ...............F........F.....................         [ 61%]
    fire/helputils_test.py .F........                                        [ 68%]
    fire/inspectutils_test.py .F.......                                      [ 73%]
    fire/interact_test.py ..                                                 [ 75%]
    fire/parser_test.py ....................                                 [ 87%]
    fire/test_components_test.py ..                                          [ 88%]
    fire/testutils_test.py ...                                               [ 90%]
    fire/trace_test.py ...............                                       [100%]

    =================================== FAILURES ===================================
    _________________________ FireTest.testFireKeywordArgs _________________________

    self = <fire.fire_test.FireTest testMethod=testFireKeywordArgs>

        def testFireKeywordArgs(self):
          self.assertEqual(
              fire.Fire(
                  tc.Kwargs,
                  command=['props', '--name', 'David', '--age', '24']),
              {'name': 'David', 'age': 24})
          # Run this test both with a list command and a string command.
          self.assertEqual(
              fire.Fire(
                  tc.Kwargs,
                  command=['props', '--message',
                           '"This is a message it has -- in it"']),  # Quotes stripped
              {'message': 'This is a message it has -- in it'})
          self.assertEqual(
              fire.Fire(
                  tc.Kwargs,
                  command=['props', '--message',
                           'This is a message it has -- in it']),
              {'message': 'This is a message it has -- in it'})
          self.assertEqual(
              fire.Fire(
                  tc.Kwargs,
                  command='props --message "This is a message it has -- in it"'),
              {'message': 'This is a message it has -- in it'})
          self.assertEqual(
              fire.Fire(tc.Kwargs,
                        command=['upper', '--alpha', 'A', '--beta', 'B']),
              'ALPHA BETA')
          self.assertEqual(
              fire.Fire(
                  tc.Kwargs,
    >             command=['upper', '--alpha', 'A', '--beta', 'B', '-', 'lower']),
              'alpha beta')

    fire/fire_test.py:240: 
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    component = <class 'fire.test_components.Kwargs'>
    command = ['upper', '--alpha', 'A', '--beta', 'B', '-', ...], name = 'py.test'

        def Fire(component=None, command=None, name=None):
          """This function, Fire, is the main entrypoint for Python Fire.
        
          Executes a command either from the `command` argument or from sys.argv by
          recursively traversing the target object `component`'s members consuming
          arguments, evaluating functions, and instantiating classes as it goes.
        
          When building a CLI with Fire, your main method should call this function.
        
          Args:
            component: The initial target component.
            command: Optional. If supplied, this is the command executed. If not
                supplied, then the command is taken from sys.argv instead. This can be
                a string or a list of strings; a list of strings is preferred.
            name: Optional. The name of the command as entered at the command line.
                Used in interactive mode and for generating the completion script.
          Returns:
            The result of executing the Fire command. Execution begins with the initial
            target component. The component is updated by using the command arguments
            to either access a member of the current component, call the current
            component (if it's a function), or instantiate the current component (if
            it's a class). When all arguments are consumed and there's no function left
            to call or class left to instantiate, the resulting current component is
            the final result.
          Raises:
            ValueError: If the command argument is supplied, but not a string or a
                sequence of arguments.
            FireExit: When Fire encounters a FireError, Fire will raise a FireExit with
                code 2. When used with the help or trace flags, Fire will raise a
                FireExit with code 0 if successful.
          """
          name = name or os.path.basename(sys.argv[0])
        
          # Get args as a list.
          if isinstance(command, six.string_types):
            args = shlex.split(command)
          elif isinstance(command, (list, tuple)):
            args = command
          elif command is None:
            # Use the command line args by default if no command is specified.
            args = sys.argv[1:]
          else:
            raise ValueError('The command argument must be a string or a sequence of '
                             'arguments.')
        
          # Determine the calling context.
          caller = inspect.stack()[1]
          caller_frame = caller[0]
          caller_globals = caller_frame.f_globals
          caller_locals = caller_frame.f_locals
          context = {}
          context.update(caller_globals)
          context.update(caller_locals)
        
          component_trace = _Fire(component, args, context, name)
        
          if component_trace.HasError():
            for help_flag in ['-h', '--help']:
              if help_flag in component_trace.elements[-1].args:
                command = '{cmd} -- --help'.format(cmd=component_trace.GetCommand())
                print(('WARNING: The proper way to show help is {cmd}.\n'
                       'Showing help anyway.\n').format(cmd=pipes.quote(command)),
                      file=sys.stderr)
        
            print('Fire trace:\n{trace}\n'.format(trace=component_trace),
                  file=sys.stderr)
            result = component_trace.GetResult()
            print(
                helputils.HelpString(result, component_trace, component_trace.verbose),
                file=sys.stderr)
    >       raise FireExit(2, component_trace)
    E       fire.core.FireExit: 2

    fire/core.py:143: FireExit
    ----------------------------- Captured stdout call -----------------------------
    name: David
    age:  24
    message: This is a message it has -- in it
    message: This is a message it has -- in it
    message: This is a message it has -- in it
    ALPHA BETA
    ----------------------------- Captured stderr call -----------------------------
    Fire trace:
    1. Initial component
    2. Instantiated class "Kwargs" (/private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source/fire/test_components.py:176)
    3. Accessed property "upper" (/private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source/fire/test_components.py:181)
    4. Called routine "upper" (/private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source/fire/test_components.py:181)
    5. Accessed property "lower"
    6. ('The function received no value for the required argument:', 'self')

    Type:        builtin_function_or_method
    String form: <built-in method lower of str object at 0x1116ffef0>
    Docstring:   Return a copy of the string converted to lowercase.

    Usage:       py.test upper --alpha A --beta B - lower SELF
                 py.test upper --alpha A --beta B - lower --self SELF
    _______________________ FireTest.testFireObjectWithDict ________________________

    self = <fire.fire_test.FireTest testMethod=testFireObjectWithDict>

        def testFireObjectWithDict(self):
          self.assertEqual(
              fire.Fire(tc.TypedProperties, command=['delta', 'echo']), 'E')
          self.assertEqual(
    >         fire.Fire(tc.TypedProperties, command=['delta', 'echo', 'lower']), 'e')

    fire/fire_test.py:269: 
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    component = <class 'fire.test_components.TypedProperties'>
    command = ['delta', 'echo', 'lower'], name = 'py.test'

        def Fire(component=None, command=None, name=None):
          """This function, Fire, is the main entrypoint for Python Fire.
        
          Executes a command either from the `command` argument or from sys.argv by
          recursively traversing the target object `component`'s members consuming
          arguments, evaluating functions, and instantiating classes as it goes.
        
          When building a CLI with Fire, your main method should call this function.
        
          Args:
            component: The initial target component.
            command: Optional. If supplied, this is the command executed. If not
                supplied, then the command is taken from sys.argv instead. This can be
                a string or a list of strings; a list of strings is preferred.
            name: Optional. The name of the command as entered at the command line.
                Used in interactive mode and for generating the completion script.
          Returns:
            The result of executing the Fire command. Execution begins with the initial
            target component. The component is updated by using the command arguments
            to either access a member of the current component, call the current
            component (if it's a function), or instantiate the current component (if
            it's a class). When all arguments are consumed and there's no function left
            to call or class left to instantiate, the resulting current component is
            the final result.
          Raises:
            ValueError: If the command argument is supplied, but not a string or a
                sequence of arguments.
            FireExit: When Fire encounters a FireError, Fire will raise a FireExit with
                code 2. When used with the help or trace flags, Fire will raise a
                FireExit with code 0 if successful.
          """
          name = name or os.path.basename(sys.argv[0])
        
          # Get args as a list.
          if isinstance(command, six.string_types):
            args = shlex.split(command)
          elif isinstance(command, (list, tuple)):
            args = command
          elif command is None:
            # Use the command line args by default if no command is specified.
            args = sys.argv[1:]
          else:
            raise ValueError('The command argument must be a string or a sequence of '
                             'arguments.')
        
          # Determine the calling context.
          caller = inspect.stack()[1]
          caller_frame = caller[0]
          caller_globals = caller_frame.f_globals
          caller_locals = caller_frame.f_locals
          context = {}
          context.update(caller_globals)
          context.update(caller_locals)
        
          component_trace = _Fire(component, args, context, name)
        
          if component_trace.HasError():
            for help_flag in ['-h', '--help']:
              if help_flag in component_trace.elements[-1].args:
                command = '{cmd} -- --help'.format(cmd=component_trace.GetCommand())
                print(('WARNING: The proper way to show help is {cmd}.\n'
                       'Showing help anyway.\n').format(cmd=pipes.quote(command)),
                      file=sys.stderr)
        
            print('Fire trace:\n{trace}\n'.format(trace=component_trace),
                  file=sys.stderr)
            result = component_trace.GetResult()
            print(
                helputils.HelpString(result, component_trace, component_trace.verbose),
                file=sys.stderr)
    >       raise FireExit(2, component_trace)
    E       fire.core.FireExit: 2

    fire/core.py:143: FireExit
    ----------------------------- Captured stdout call -----------------------------
    E
    ----------------------------- Captured stderr call -----------------------------
    Fire trace:
    1. Initial component
    2. Instantiated class "TypedProperties" (/private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source/fire/test_components.py:98)
    3. Accessed property "delta"
    4. Accessed property "echo"
    5. Accessed property "lower"
    6. ('The function received no value for the required argument:', 'self')

    Type:        builtin_function_or_method
    String form: <built-in method lower of str object at 0x1100c7340>
    Docstring:   Return a copy of the string converted to lowercase.

    Usage:       py.test delta echo lower SELF
                 py.test delta echo lower --self SELF
    _____________________ HelpUtilsTest.testHelpStringBuiltin ______________________

    self = <fire.helputils_test.HelpUtilsTest testMethod=testHelpStringBuiltin>

        def testHelpStringBuiltin(self):
          helpstring = helputils.HelpString('test'.upper)
          self.assertIn('Type:        builtin_function_or_method', helpstring)
          self.assertIn('String form: <built-in method upper of', helpstring)
    >     self.assertIn('Usage:       [VARS ...] [--KWARGS ...]', helpstring)
    E     AssertionError: 'Usage:       [VARS ...] [--KWARGS ...]' not found in 'Type:        builtin_function_or_method\nString form: <built-in method upper of str object at 0x11028d458>\nDocstring:   Return a copy of the string converted to uppercase.\n\nUsage:       SELF\n             --self SELF'

    fire/helputils_test.py:87: AssertionError
    ________________ InspectUtilsTest.testGetFullArgSpecFromBuiltin ________________

    self = <fire.inspectutils_test.InspectUtilsTest testMethod=testGetFullArgSpecFromBuiltin>

        def testGetFullArgSpecFromBuiltin(self):
          spec = inspectutils.GetFullArgSpec('test'.upper)
    >     self.assertEqual(spec.args, [])
    E     AssertionError: Lists differ: ['self'] != []
    E     
    E     First list contains 1 additional elements.
    E     First extra element 0:
    E     'self'
    E     
    E     - ['self']
    E     + []

    fire/inspectutils_test.py:57: AssertionError
    =============================== warnings summary ===============================
    examples/diff/diff_test.py::DiffTest::testContextDiff
    examples/diff/diff_test.py::DiffTest::testDiffFull
    examples/diff/diff_test.py::DiffTest::testMakeDiff
    examples/diff/diff_test.py::DiffTest::testNDiff
    examples/diff/diff_test.py::DiffTest::testSetUp
    examples/diff/diff_test.py::DiffTest::testUnifiedDiff
      /private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source/examples/diff/diff.py:77: DeprecationWarning: 'U' mode is deprecated
        self.fromlines = open(fromfile, 'U').readlines()

    examples/diff/diff_test.py::DiffTest::testContextDiff
    examples/diff/diff_test.py::DiffTest::testDiffFull
    examples/diff/diff_test.py::DiffTest::testMakeDiff
    examples/diff/diff_test.py::DiffTest::testNDiff
    examples/diff/diff_test.py::DiffTest::testSetUp
    examples/diff/diff_test.py::DiffTest::testUnifiedDiff
      /private/var/folders/8g/qgvf93_s1_b40dd8nhsvqk_m0000gn/T/nix-build-python3.7-fire-0.1.3.drv-0/source/examples/diff/diff.py:78: DeprecationWarning: 'U' mode is deprecated
        self.tolines = open(tofile, 'U').readlines()

    -- Docs: https://docs.pytest.org/en/latest/warnings.html
    ============== 4 failed, 156 passed, 12 warnings in 1.80 seconds ===============
    builder for '/nix/store/dpcwgrlydrrlbdrnf0mi9hmr0kjvia89-python3.7-fire-0.1.3.drv' failed with exit code 1
    error: build of '/nix/store/dpcwgrlydrrlbdrnf0mi9hmr0kjvia89-python3.7-fire-0.1.3.drv' failed

## Technical details

nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.5.0, macOS 10.14.4`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2`
 - channels(abathur): `"nixpkgs-19.09pre174594.0c0954781e2, darwin"`
 - nixpkgs: `/Users/abathur/.nix-defexpr/channels/nixpkgs`

FWIW, I'm also running bash 5 (`5.0.2(1)-release`) as my default shell atm, but I see the same error when I manually run launch a bash 4 (`4.4.23(1)-release`) shell to run this in. I haven't updated channel in a bit but I don't see any open/recent issue or PR.

--- a/pkgs/development/python-modules/fire/default.nix
+++ b/pkgs/development/python-modules/fire/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, buildPythonPackage, fetchFromGitHub, six, hypothesis, mock
+{ stdenv, buildPythonPackage, fetchFromGitHub, fetchpatch, six, hypothesis, mock
 , python-Levenshtein, pytest }:
 
 buildPythonPackage rec {
@@ -20,6 +20,13 @@ buildPythonPackage rec {
     py.test
   '';
 
+  patches = [
+    # Add Python 3.7 support. Remove with the next release
+    (fetchpatch {
+      url = "https://github.com/google/python-fire/commit/668007ae41391f5964870b4597e41493a936a11e.patch";
+      sha256 = "0rf7yzv9qx66zfmdggfz478z37fi4rwx4hlh3dk1065sx5rfksi0";
+    })
+  ];
 
   meta = with stdenv.lib; {
     description = "A library for automatically generating command line interfaces";

